blockindex/contractstaking/indexer_test.go

```diff
@@ -351,6 +351,23 @@ func TestContractStakingIndexerBucketInfo(t *testing.T) {
 	r.EqualValues(1, indexer.TotalBucketCount())
 }
 
+func BenchmarkIndexer_PutBlockBeforeContractHeight(b *testing.B) {
+	// Create a new Indexer with a contract height of 100
+	indexer := &Indexer{contractHeight: 100}
+
+	// Create a mock block with a height of 50
+	blk := &block.Block{}
+
+	// Run the benchmark
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		err := indexer.PutBlock(context.Background(), blk)
+		if err != nil {
+			b.Fatal(err)
+		}
+	}
+}
+
```

Comment:
The Benchmark Report:
```
❯ go test -benchmem -run=^$ -bench ^BenchmarkIndexer_PutBlockBeforeContractHeight$ github.com/iotexproject/iotex-core/blockindex/contractstaking
goos: darwin
goarch: arm64
pkg: github.com/iotexproject/iotex-core/blockindex/contractstaking
BenchmarkIndexer_PutBlockBeforeContractHeight-10        522791816                2.249 ns/op           0 B/op          0 allocs/op
PASS
ok      github.com/iotexproject/iotex-core/blockindex/contractstaking   1.687s
```

Currently, the mainnet height is approximately 23878680. The estimated time for the startup check is `23878680 * 2.249 ns = 0.0537 s`. This result is acceptable